django/core/files/storage.py

```diff
@@ -324,6 +339,25 @@ def created_time(self, name):
     def modified_time(self, name):
         return datetime.fromtimestamp(os.path.getmtime(self.path(name)))
 
+    def _do_safe_operation(self, operation, src_path, dst_path):
+        try:
+            operation(src_path, dst_path)
+        except OSError as e:
+            if e.errno == errno.EEXIST:
+                raise ValueError('Destination already exists')
```

Comment:
I'm not sure about `FileExistsError` either. `FileExistsError` also a subclass of `OSError` which brings us back to the same problem: is not being able to move a file on S3 an `OSError`?